deps/jerry/jerry-core/CMakeLists.txt

```diff
@@ -17,27 +17,28 @@ set(JERRY_CORE_NAME jerry-core)
-set(FEATURE_SOURCE_INFO        ON       CACHE BOOL   "Enable saving Javascript source info?")
-set(FEATURE_PARSER_DUMP        OFF      CACHE BOOL   "Enable parser byte-code dumps?")
-set(FEATURE_PROFILE            "es5.1"  CACHE STRING "Use default or other profile?")
-set(FEATURE_REGEXP_STRICT_MODE OFF      CACHE BOOL   "Enable regexp strict mode?")
-set(FEATURE_REGEXP_DUMP        OFF      CACHE BOOL   "Enable regexp byte-code dumps?")
-set(FEATURE_SNAPSHOT_EXEC      OFF      CACHE BOOL   "Enable executing snapshot files?")
-set(FEATURE_SNAPSHOT_SAVE      OFF      CACHE BOOL   "Enable saving snapshot files?")
-set(FEATURE_SYSTEM_ALLOCATOR   OFF      CACHE BOOL   "Enable system allocator?")
-set(FEATURE_VALGRIND           OFF      CACHE BOOL   "Enable Valgrind support?")
-set(FEATURE_VALGRIND_FREYA     OFF      CACHE BOOL   "Enable Valgrind-Freya support?")
-set(FEATURE_VM_EXEC_STOP       OFF      CACHE BOOL   "Enable VM execution stopping?")
-set(MEM_HEAP_SIZE_KB           "5120"   CACHE STRING "Size of memory heap, in kilobytes")
+set(FEATURE_CPOINTER_32_BIT       ON       CACHE BOOL   "Enable 32 bit compressed pointers?")
+set(FEATURE_DEBUGGER              ON       CACHE BOOL   "Enable JerryScript debugger?")
+set(FEATURE_ERROR_MESSAGES        ON       CACHE BOOL   "Enable error messages?")
+set(FEATURE_EXTERNAL_CONTEXT      OFF      CACHE BOOL   "Enable external context?")
+set(FEATURE_JS_PARSER             ON       CACHE BOOL   "Enable js-parser?")
+set(FEATURE_MEM_STATS             OFF      CACHE BOOL   "Enable memory statistics?")
+set(FEATURE_MEM_STRESS_TEST       OFF      CACHE BOOL   "Enable mem-stress test?")
+set(FEATURE_CPU_PROFILER          OFF      CACHE BOOL   "Enable Cpu profiler?")
+set(FEATURE_HEAP_PROFILER         OFF      CACHE BOOL   "Enable Heap profiler?")
+set(FEATURE_SOURCE_INFO           ON       CACHE BOOL   "Enable saving Javascript source info?")
+set(FEATURE_PARSER_DUMP           OFF      CACHE BOOL   "Enable parser byte-code dumps?")
+set(FEATURE_PROFILE               "es5.1"  CACHE STRING "Use default or other profile?")
+set(FEATURE_REGEXP_STRICT_MODE    OFF      CACHE BOOL   "Enable regexp strict mode?")
+set(FEATURE_REGEXP_DUMP           OFF      CACHE BOOL   "Enable regexp byte-code dumps?")
+set(FEATURE_SNAPSHOT_EXEC         OFF      CACHE BOOL   "Enable executing snapshot files?")
+set(FEATURE_SNAPSHOT_SAVE         OFF      CACHE BOOL   "Enable saving snapshot files?")
+set(FEATURE_SYSTEM_ALLOCATOR      OFF      CACHE BOOL   "Enable system allocator?")
+set(FEATURE_VALGRIND              OFF      CACHE BOOL   "Enable Valgrind support?")
+set(FEATURE_VALGRIND_FREYA        OFF      CACHE BOOL   "Enable Valgrind-Freya support?")
+set(FEATURE_VM_EXEC_STOP          OFF      CACHE BOOL   "Enable VM execution stopping?")
+set(MEM_HEAP_SIZE_KB              "5120"   CACHE STRING "Size of memory heap, in kilobytes")
+set(HEAP_MAX_SIZE_FACTOR          "1"      CACHE STRING "Above CONFIG_MEM_HEAP_MAX_LIMIT value")
```

Comment:
Suggestion: "expand heap max size limit to CONFIG_MEM_HEAP_MAX_LIMIT multiplied by the factor"